src/main/java/marquez/service/DatasourceService.java

```diff
@@ -43,10 +43,14 @@ public Datasource create(@NonNull ConnectionUrl connectionUrl, @NonNull Datasour
     final DatasourceRow row = DatasourceRowMapper.map(connectionUrl, name);
 
     try {
-      datasourceDao.insert(row);
-      final DatasourceUrn urn = DatasourceUrn.fromString(row.getUrn());
-      final Optional<DatasourceRow> rowIfFound = datasourceDao.findBy(urn.getValue());
-      return rowIfFound.map(DatasourceMapper::map).orElseThrow(MarquezServiceException::new);
+      // Check if its already there based on its name. If so, return
+      final Optional<DatasourceRow> existingRowIfFound = datasourceDao.findBy(name);
```

Comment:
I think we can use `DatasourceService.get(DatasourceUrn)`? You have everything needed to build the URN.